python/cudf/dataframe/series.py

```diff
@@ -1833,11 +1833,47 @@ def quantile(self, q=0.5, interpolation='linear', exact=True,
         DataFrame
 
         """
+        def invalid_quantile(q):
+            if 0 <= q <= 1:
+                return False
+            else:
+                return True
+
+        if isinstance(q, Number) and invalid_quantile(q):
+            raise ValueError("percentiles should all \
+                             be in the interval [0, 1]")
+
+        if utils.is_list_like(q):
+            invalid_q_vector = np.vectorize(invalid_quantile,
+                                            otypes=[np.bool8])
```

Comment:
This isn't what I mean't. You should be able to just do `np.logical_or(q >= 0, q <= 1).all()` which will return a scalar True if the quantiles are valid.

Reply:
Okay, modified it. But doing an or on invalid q's and performing an `any` on them so that way we don't have to scan entire `q` when all q's are invalid.